great_expectations/cli/toolkit.py

```diff
@@ -411,8 +437,114 @@ def load_data_context_with_error_handling(directory: str) -> DataContext:
         sys.exit(1)
 
 
+def upgrade_project(
+    context_root_dir, ge_config_version, from_cli_upgrade_command=False
+):
+    continuation_message = (
+        "\nOk, exiting now. To upgrade at a later time, use the following command: "
+        "<cyan>great_expectations project upgrade</cyan>\n\nTo learn more about the upgrade "
+        "process, visit "
+        "<cyan>https://docs.greatexpectations.io/en/latest/how_to_guides/migrating_versions.html"
+        "</cyan>.\n"
+    )
+    if from_cli_upgrade_command:
+        message = (
+            f"<red>\nYour project appears to have an out-of-date config version ({ge_config_version}) - "
+            f"the version "
+            f"number must be at least {MINIMUM_SUPPORTED_CONFIG_VERSION}.</red>"
+        )
+    else:
+        message = (
+            f"<red>\nYour project appears to have an out-of-date config version ({ge_config_version}) - "
+            f"the version "
+            f"number must be at least {MINIMUM_SUPPORTED_CONFIG_VERSION}.\nIn order to proceed, "
+            f"your project must be upgraded.</red>"
+        )
+
+    cli_message(message)
+    upgrade_prompt = (
+        "\nWould you like to run the Upgrade Helper to bring your project up-to-date?"
+    )
+    confirm_proceed_or_exit(
+        confirm_prompt=upgrade_prompt, continuation_message=continuation_message
+    )
+    cli_message(SECTION_SEPARATOR)
+
+    # use loop in case multiple upgrades need to take place
+    while ge_config_version < MINIMUM_SUPPORTED_CONFIG_VERSION:
+        upgrade_helper_class = GE_UPGRADE_HELPER_VERSION_MAP.get(int(ge_config_version))
+        if not upgrade_helper_class:
+            break
+        target_ge_config_version = int(ge_config_version) + 1
```

Comment:
This is super clever--love it! Only thing to think about (not for now) would be to allow config versions to also have some sort of major/minor concept (e.g. version 1.1)

Reply:
Oh, that was the reasoning around coercing to int. Figured we might have say 1.1 versions, etc., but would only have UpgradeHelpers for major increments, i.e. 1 to 2.